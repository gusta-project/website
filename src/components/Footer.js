import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';
import { Link } from 'gatsby-plugin-i18next';

class Footer extends Component {
  static propTypes = {
    siteTitle: PropTypes.string.isRequired,
    t: PropTypes.func.isRequired
  };

  static defaultProps = {
    siteTitle: ''
  };

  render() {
    const { t } = this.props;

    return (
      <footer className="mt-5 text-right">
        <p>
          &copy; {new Date(Date.now()).getFullYear()} {this.props.siteTitle}
          &nbsp;
        </p>
        <Link to="/about">{t('about_us')}</Link> |&nbsp;
        <Link to="/privacy">{t('privacy_policy')}</Link> | {t('license_stmt')}
        <a href="https://opensource.org/licenses/MIT">MIT License</a>
      </footer>
    );
  }
}

export default translate()(Footer);
