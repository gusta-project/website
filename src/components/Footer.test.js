import React from 'react';
import renderer from 'react-test-renderer';

import Footer from './Footer';

Date.now = jest.fn(() => new Date(2020, 0, 1, 12));

describe('Footer', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Footer siteTitle="Footer Test" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
