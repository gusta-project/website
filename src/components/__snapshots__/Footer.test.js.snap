// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  className="mt-5 text-right"
>
  <p>
    © 
    2019
     
    Footer Test
     
  </p>
  <Link
    to="/about"
  >
    about_us
  </Link>
   | 
  <Link
    to="/privacy"
  >
    privacy_policy
  </Link>
   | 
  license_stmt
  <a
    href="https://opensource.org/licenses/MIT"
  >
    MIT License
  </a>
</footer>
`;
